external modules can add items to the list of menu items to hide when…

stuartromanek opened the pull request:
… in preview mode

anything to add here?

boutell commented on lib/modules/apostrophe-workflow-admin-bar/index.js:
```diff
@@ -17,6 +17,9 @@ module.exports = {
         return result;
       }
       var notSafeLiveList = [ 'apostrophe-pages', 'apostrophe-tags', 'apostrophe-workflow-manage-modal' ];
```
Our pattern in other places and modules has been "baseNotSafeLiveList" and "notSafeLiveList," the latter being the one that gets concatenated to the base. The base gets set up as an option also, so that if someone is really on fire they may override it.

boutell replied:
Also in yesterday's engineering huddle we agreed to use the terms "allowList" and "denyList" in such contexts (where each is actually present - we didn't say both had to exist).

stuartromanek commented:
changed option name to `removeWhenLive` per internal discussion .. I also went ahead and softly concat'd the defaults and removed the need for a `base` .. does that feel alright @boutell ?

boutell commented on CHANGELOG.md:
```diff
@@ -1,5 +1,8 @@
 # Changelog
 
+## 2.37.0
+* Adds `options.baseSafeLiveDenyList` to `apostrophe-admin-bar`, a list of items to be removed from the admin bar during preview/live mode. It can be added to from other `apostrophe-admin-bar` improvers by including an array of apostrophe-module names in `options.safeLiveDenyList`.
```
Doc needs an update to match.
